src/backend/columnar/columnar_tableam.c

```diff
@@ -2454,7 +2515,11 @@ alter_columnar_table_reset(PG_FUNCTION_ARGS)
 							   quote_identifier(RelationGetRelationName(rel)))));
 	}
 
-	EnsureTableOwner(relationId);
+	if (!pg_class_ownercheck(relationId, GetUserId()))
+	{
+		aclcheck_error(ACLCHECK_NOT_OWNER, OBJECT_TABLE,
+					   get_rel_name(relationId));
```

Comment:
You can add a SQL test so that this code is covered.

Reply:
Added in multi_multiuser.sql